stdlib/String.tr

```diff
@@ -1,76 +1,80 @@
 type String {
+    // Underlying storage for String.
+    // - note: MUST always end in a NUL terminator.
     var _storage: ByteArray
     init(cString: *Int8) {
         self._storage = ByteArray(cString)
         self._storage.append(0)
     }
-    init(_owning cString: *Int8, length: Int) {
-        self._storage = ByteArray(_owning: cString, length: length)
-    }
     init(_global cString: *Int8, length: Int) {
         // TODO: make this not call into a ByteArray initializer
         // that immediately allocates a new buffer -- it should be
         // CoW
         self._storage = ByteArray(cString, length: length)
+        self._storage.append(0)
```

Comment:
Yes. We want to guarantee it will be NUL-terminated to avoid making heap-allocations on calls to `cString`.